ztp/siteconfig-generator/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+
+	siteConfigs "github.com/openshift-kni/cnf-features-deploy/ztp/siteconfig-generator/siteConfig"
+	"github.com/spf13/pflag"
+	"gopkg.in/yaml.v3"
+)
+
+func main() {
+	// Parse command input
+	pflag.Parse()
+
+	// Collect and parse siteconfig files paths
+	siteConfigFiles := pflag.Args()
+	var outputBuffer bytes.Buffer
+	scBuilder, _ := siteConfigs.NewSiteConfigBuilder()
+
+	for _, siteConfigFile := range siteConfigFiles {
+		fileData, err := siteConfigs.ReadFile(siteConfigFile)
+		if err != nil {
+			log.Printf("Error: could not read file %s: %s\n", siteConfigFile, err)
+			continue
+		}
+
+		siteConfig := siteConfigs.SiteConfig{}
+		err = yaml.Unmarshal(fileData, &siteConfig)
+		if err != nil {
+			log.Printf("Error: could not parse %s as yaml: %s\n", siteConfigFile, err)
+			continue
+		}
+
+		clusters, err := scBuilder.Build(siteConfig)
+		if err != nil {
+			log.Printf("Error: could not build the entire SiteConfig defined by %s: %s", siteConfigFile, err)
+			continue
+		}
+		for _, crs := range clusters {
+			for _, crIntf := range crs {
+				cr, err := yaml.Marshal(crIntf)
+				if err != nil {
+					panic(err)
```

Comment:
Why panic here when other errors are logged and continued? Maybe clear out the outputBuffer and break out of this loop so that no CRs for this cluster are generated but other clusters can continue.